packages/alea-frontend/pages/blog/[postId].tsx

```diff
@@ -20,20 +20,36 @@ import {
   isModerator,
 } from '@stex-react/api';
 import { MdViewer } from '@stex-react/markdown';
-import { GetServerSidePropsContext, NextPage } from 'next';
+import fs from 'fs';
+import { NextPage } from 'next';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import MainLayout from '../../layouts/MainLayout';
 
 const BlogPostPage: NextPage = ({ post }: { post: BlogPost }) => {
+  const router = useRouter();
+  const postId = router.query.postId;
   const [userInfo, setUserInfo] = useState<UserInfo>(null);
   const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
-  const router = useRouter();
+  const [blogPost, setBlogPost] = useState<BlogPost>(post);
 
   useEffect(() => {
-    getUserInfo().then(setUserInfo);
-  }, []);
+    async function fetchBlog() {
+      try {
+        const userInfoData = await getUserInfo();
+        const blogData = await getPostById(postId as string);
+        setUserInfo(userInfoData);
+        setBlogPost(blogData);
+      } catch (error) {
+        alert('the page youe are looking for is not available');
+        router.push('/blog');
+      }
+    }
+    if (router.isReady && postId) {
+      fetchBlog();
+    }
+  }, [router.isReady, postId, router.isFallback, router]);
```

Comment:
Why is `isFallback` in this list? You are always calling `getPostId` (in both fallback and not fallback scenarios). Can you skip the call in non-fallback scenario with revalidation?